src/commands/publish.rs

```diff
@@ -51,7 +55,17 @@ pub fn publish(
             delete_bulk(target, user, &site_namespace.id, to_delete)?;
         }
     } else {
-        upload::script(&user, &target, None)?;
+        let uses_kv_bucket = sync_non_site_buckets(target, user, verbose)?;
+
+        let feature = if uses_kv_bucket {
+            message::deprecation_warning("As of 1.9.0, you will no longer be able to specify a `bucket` for a kv namespace in your `wrangler.toml`. If your application depends on this feature, please file an issue at https://github.com/cloudflare/wrangler/issues/new with your use case.https://github.com/cloudflare/wrangler/issues/new?labels=user+report&template=feature_request.md");
```

Comment:
this could probably use some whitespace and maybe a bit of coloring. in #1128 i use style(url).blue().bold() for urls and style(code).yellow().bold() for code blocks (like `wrangler.toml`)

```suggestion
            message::deprecation_warning("As of 1.9.0, you will no longer be able to specify a `bucket` for a kv namespace in your `wrangler.toml`.\nIf your application depends on this feature, please file an issue with your use case here:\nhttps://github.com/cloudflare/wrangler/issues/new?labels=user+report&template=feature_request.md");
```